django/db/backends/sqlite3/features.py

```diff
@@ -44,6 +43,20 @@ class DatabaseFeatures(BaseDatabaseFeatures):
     supports_order_by_nulls_modifier = Database.sqlite_version_info >= (3, 30, 0)
     order_by_nulls_first = True
 
+    @cached_property
+    def supports_atomic_references_rename(self):
+        if Database.sqlite_version_info < (3, 26, 0):
+            return False
+
```

Comment:
Could we possibly skip the detection feature `if Database.sqlite_version_info >= (3, 29, 0)`? Since this is only present on macOS [could we branch off `platform.platform`](https://docs.python.org/3/library/platform.html?highlight=darwin#platform.platform)? I think this would partially address @claudep's concerns.

Reply:
We can do this, but I feel that it's not correct. This code *should* work with 3.29.0 but there's clearly a configuration option or a bug that prevents it from working. As such, are we confident that `>=(3, 29, 0)` will always work?

For the same reason I'd be against branching of `platform.platform`. I don't think the aim here should be to "detect this specific MacOS version of SQLite" but should be more generally "Detect if sqlite _actually supports_ renaming atomic references". I don't see how we can do that without feature detection.

Reply:
In an ideal world, yes. But I don't have the time to deep-dive into this, all I can say for sure is that the default sqlite library on MacOS doesn't exhibit the expected behaviour across the three machines I've tried it on. I'm not sure if it's a bug or an explicit decision made by Apple for backwards compatibility reasons. 

I'd suggest it might be the latter, and that they probably maintain an internal fork of SQLite with custom patches rather than the "vanilla" distribution. So we can either:

1. exclude sqlite3 `3.28.0` when running on MacOS `10.15.5` 
2. use feature detection
3. deep dive into the MacOS XNU sources they publish, or otherwise report it to apple via RADAR, and see what they say

Honestly I'm not _that_ bothered about which direction we take - this whole thing spun out of me trying to get the GSOC branch passing on my local machine 😭. I'm OK to exclude MacOS 10.15.x from this feature, but in case anyone in the future re-visits this because a future version of MacOS also don't work with this feature, I'd like to register my dislike!